codec-registration-checker-compiler-plugin/src/main/scala/org/virtuslab/ash/SerializerCheckCompilerPluginComponent.scala

```diff
@@ -176,6 +160,69 @@ class SerializerCheckCompilerPluginComponent(
             None
         }
       }
+
+      private def collectMissingClassNames(fullyQualifiedClassNames: List[String]): List[String] = {
+        val sourceCodeDir = File(options.sourceCodeDirectoryToCheck)
+        val sourceCodeFilesAsStrings =
+          (for (file <- sourceCodeDir.collectChildren(_.name.endsWith(".scala"))) yield file.contentAsString).toList
+
+        def typeIsDefinedInScalaFiles(fqcn: String): Boolean = {
+          val indexOfLastDotInFQCN = fqcn.lastIndexOf('.')
+          val packageName = fqcn.substring(0, indexOfLastDotInFQCN)
+          val typeName = fqcn.substring(indexOfLastDotInFQCN + 1)
+          sourceCodeFilesAsStrings.exists(fileAsString => {
+            fileAsString.startsWith(s"package $packageName") &&
+            (fileAsString.contains(s"class $typeName") || fileAsString.contains(s"trait $typeName") || fileAsString
+              .contains(s"object $typeName"))
+          })
+        }
+
+        fullyQualifiedClassNames.filter(typeIsDefinedInScalaFiles)
+      }
+
+      private def interactWithTheCacheFile(
+          mode: CacheFileInteractionMode,
+          typeNamesToRemove: List[String] = List.empty): Unit =
+        mode match {
+          case _: DumpTypesIntoCacheFile =>
```

Comment:
In a nutshell, this case is the old piece from `def apply`, while the other case is newly implemented, right?

Reply:
yes